comp110-worksheet-7/DirectoryUtils.cs

```diff
@@ -24,37 +24,204 @@ public static bool IsDirectory(string path)
 		// Return the total size, in bytes, of all the files below the given directory
 		public static long GetTotalSize(string directory)
 		{
-			throw new NotImplementedException();
+            long directorySize = 0;
+            Stack<string> directoriesLeft = new Stack<string>();
+
+            if (IsDirectory(directory))
+            {
+                directoriesLeft.Push(directory);
+
+                while (directoriesLeft.Count > 0)
+                {
+                    string currentDirectory = directoriesLeft.Pop();
+                    string[] newDirectories = Directory.GetDirectories(currentDirectory);
+                    string[] newfiles = Directory.GetFiles(currentDirectory);
+
+                    for (int i = 0; i < newDirectories.Length; i++)
+                    {
+                        directoriesLeft.Push(newDirectories[i]);
+                    }
+
+                    for (int i = 0; i < newfiles.Length; i++)
+                    {
+                        directorySize += GetFileSize(newfiles[i]);
+                    }
+                }
+            }
+            return directorySize;
 		}
 
 		// Return the number of files (not counting directories) below the given directory
 		public static int CountFiles(string directory)
 		{
-			throw new NotImplementedException();
-		}
+            int directoryCount = 0;
+            Stack<string> directoriesLeft = new Stack<string>();
+
+            if (IsDirectory(directory))
+            {
+                directoriesLeft.Push(directory);
+
+                while (directoriesLeft.Count > 0)
+                {
+                    string currentDirectory = directoriesLeft.Pop();
+                    string[] newDirectories = Directory.GetDirectories(currentDirectory);
+                    string[] newfiles = Directory.GetFiles(currentDirectory);
+
+                    for (int i = 0; i < newDirectories.Length; i++)
+                    {
+                        directoriesLeft.Push(newDirectories[i]);
+                    }
+
+                    for (int i = 0; i < newfiles.Length; i++)
+                    {
+                        directoryCount++;
+                    }
+                }
+            }
+            return directoryCount;
+        }
 
-		// Return the nesting depth of the given directory. A directory containing only files (no subdirectories) has a depth of 0.
+		// Return the nesting depth of the given directory. A directory containing only files (no currentDirectories) has a depth of 0.
```

Comment:
By any chance did you use the Visual Studio refactor tool to rename a variable? Looks like it changed the text of the comment too -- be careful of this!